### ISSUE TITLE
Make it possible to start/stop/restart single services

### ISSUE THREAD

**Issue by jooooel** [dotnet/tye#876]
### What should we add or change to make your life better?
I'm using Tye to run a couple of infrastructure containers (SEQ, Postgres, Cassandra, RabbitMQ, etc), a couple of our services as containers and the service(s) I'm currently working on as csproj(s). About ~20 services all in all.

I jump a lot between branches (code reviews, fixes here and there, etc) and most of the time I want to keep a base set of containers running sine they hardly ever change between revisions. But, I would like to be able to stop, start and restart specific services (both containers and csprojs) on demand.

Restarting Tye entirely can take a long time, when maybe all I really want is to restart a single service or change the version of an image. I have tried to use `watch` where that is applicable, but unfortunately that causes VS to hang depending on what parts of the code that changes (a common lib for example). And it doesn't work when changing image versions.

I think there are a couple of different commands that would help me in my daily work. All of these would require that Tye can run in detached mode:

1. `tye run <service name>`. Would make it possible to add a specific service to the current Tye instance.
2. `tye stop <service name>`. Would stop a specific service.
3. `tye restart <service name>`. Would restart a specific service.

I wanted to see how feasible this is with the current architecture of Tye, so I did a small PoC where I made it possible to restart a single (csproj) service from the Dashboard. [You can have a look at it here](https://github.com/jooooel/tye/pull/1/files). It seems to be working fine, but would require a bit of refactoring, it's not the prettiest solution at the moment.

I would love to spend more time on this and make a proper PR, and I really think it would add a lot of value.

### Why is this important to you?
Tye has really simplified my daily workflow, in that it lets me run a combination of containers and csprojs together. I also really like that I can configure them and easily decide what to run in a single file. 

Unfortunately, jumping around between branches forces me to restart Tye entirely between each run, which takes a lot of time.

**Comment by jkotalik** [comment]
Thanks for the feedback @jooooel . Your scenarios is definitely not unique, and the suggestion you have seems promising.

We have a few ideas around improving the inner loop speed in tye, especially around running multiple times. One idea we are thinking about is a daemon mode where tye is always running and commands to tye run would just add them to the daemon. That way in your example, you could arbitrarily start and stop specific services accordingly. This also would allow you to have things always running in the background without having them restart on each execution of tye run.

I could see it looking like this:
- `tye run` is the same as today, ctrl-C to kill all things that are started with tye run
- `tye start` does tye run but returns after all projects/images are started
- `tye stop` stops images and projects that are already running
- `tye restart` does the restart

Also, all of these could be specified for a specific service.

**Comment by jooooel** [comment]
Yeah that sounds pretty much like what I was hoping for as well @jkotalik.

The implementation I did was just to get a feel for how Tye is built and to get a picture of what it would take to implement those kind of changes. 

I guess it would even be a necessity if Tye is ever going to be hooked up to dev tools like VS and VS code right?

**Comment by brunolauze** [comment]
The first thing required is minimally a switch on restarting services to avoid the thing to just spin out of control when failing.

restart: no or restart: 3

... or maybe there's a way to do it ?

**Comment by jooooel** [comment]
> One idea we are thinking about is a daemon mode where tye is always running and commands to tye run would just add them to the daemon. 

I’m thinking of taking a stab at this. Mostly for fun, but because it would help me in my daily work. I’m curious of how, in your opinion, this "daemon mode" should be designed @jkotalik?

It’s great that Tye is a simple CLI tool, so having to install something like a Windows Service/systemd doesn’t feel like an option.

I guess `tye run` would have to start the host in a separate background process so that it’ll stay alive between runs right? 

Then the host would have to expose an API (over HTTP?) that can accept new commands while it’s running.

Would be happy to hear your (or anyone else’s) thoughts on this!

**Comment by malisancube** [comment]
This would be a very nice feature. I've found that using tye, I need to continuously stop all services to see the new updates.
